src/audio/rtnr/rtnr.c

```diff
@@ -399,26 +413,66 @@ static int rtnr_cmd_get_data(struct comp_dev *dev,
 	return ret;
 }
 
+static int rtnr_set_bin_data(struct comp_dev *dev,
+				       struct sof_ipc_ctrl_data *cdata)
+{
+	struct comp_data *cd = comp_get_drvdata(dev);
+	int ret = 0;
+
+	assert(cd);
+
+	comp_dbg(dev, "rtnr_set_bin_data() msg_index = %d, num_elems = %d, remaining = %d ",
+		 cdata->msg_index, cdata->num_elems,
+		 cdata->elems_remaining);
+
+	switch (cdata->data->type) {
+	case SOF_RTNR_CONFIG:
+		comp_dbg(dev, "rtnr_set_bin_data(): SOF_RTNR_CONFIG");
+		ret = comp_data_blob_set_cmd(cd->model_handler, cdata);
```

Comment:
Currently the model_handler is used for settings only( sample rate/ on&off ). For other data we would like to modify the internal data in RTNR directly.

Reply:
@jsarha is updated this API at, @jsarha can we support multiple large blobs per module/component ?
@MingJenTai I think the data would have to be serialized so that we only send one at a time from the host, @jsarha pls correct me if I'm wrong.